crates/partition-store/tests/outbox_table_test/mod.rs

```diff
@@ -58,14 +75,42 @@ pub(crate) async fn verify_outbox_is_empty_after_truncation<T: OutboxTable>(txn:
     assert_eq!(result, None);
 }
 
+pub(crate) async fn verify_neighbor_partitions_unchanged<T: OutboxTable>(txn: &mut T) {
+    let partition1336 = PartitionId::from(1336);
+    let predecessor_partition = txn
+        .get_next_outbox_message(partition1336, 0)
+        .await
+        .expect("should not fail");
+
+    let partition1338 = PartitionId::from(1338);
+    let successor_partition = txn
+        .get_next_outbox_message(partition1338, 0)
+        .await
+        .expect("should not fail");
+
+    assert!(successor_partition.is_some());
+    assert!(predecessor_partition.is_some());
+}
+
 pub(crate) async fn run_tests(mut rocksdb: PartitionStore) {
     let mut txn = rocksdb.transaction();
```

Comment:
These transactions look a bit clumsy; I tried to create a helper that takes a transaction body as a closure but it ends up being quite involved as the type isn't object-safe, requiring a bunch of extra scaffolding. It's not clear to me that we benefit from committing in the interim but it is more representative of real world use.